From: SeongJae Park <sj@kernel.org>
Date: Fri, 25 Apr 2025 12:04:37 -0700
Subject: [PATCH] Revert "mm/mm_init: use for_each_valid_pfn() in
 init_unavailable_range()"

This reverts commit b8e6d2d65f30570ecd1b6530b90f7e74bbd0f6a6.

The commit makes boot fail.  Revert now as a temporal fixup.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/mm_init.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/mm/mm_init.c b/mm/mm_init.c
index baf3cc993767..a8ca5534fc54 100644
--- a/mm/mm_init.c
+++ b/mm/mm_init.c
@@ -851,7 +851,11 @@ static void __init init_unavailable_range(unsigned long spfn,
 	unsigned long pfn;
 	u64 pgcnt = 0;
 
-	for_each_valid_pfn(pfn, spfn, epfn) {
+	for (pfn = spfn; pfn < epfn; pfn++) {
+		if (!pfn_valid(pageblock_start_pfn(pfn))) {
+			pfn = pageblock_end_pfn(pfn) - 1;
+			continue;
+		}
 		__init_single_page(pfn_to_page(pfn), pfn, zone, node);
 		__SetPageReserved(pfn_to_page(pfn));
 		pgcnt++;
-- 
2.39.5

